libdnf5-cli/argument_parser.cpp

```diff
@@ -47,6 +48,41 @@ std::size_t replace_all(std::string & inout, std::string_view what, std::string_
     return count;
 }
 
+// Function to replace occurrences of ${<index>} with strings from the array
+std::string format_string(const std::string & format, int argc, const char * const argv[]) {
+    const std::regex placeholder(R"(\$\{(\d+)\})");
+
+    std::string formatted;
+
+    // Search for all occurrences of ${<index>}
+    auto begin = format.cbegin();
+    auto end = format.cend();
+    std::smatch match;
+    while (std::regex_search(begin, end, match, placeholder)) {
+        formatted.append(begin, match[0].first);  // Append text before the placeholder
+        int index;
+        try {
+            index = std::stoi(match[1]);
+        } catch (const std::invalid_argument & ex) {
+            throw ArgumentParserInvalidValueError(
+                M_("Misconfigured alias \"{}\": cannot convert \"{}\" to a number"),
+                std::string(argv[0]),
+                match[0].str());
+        } catch (const std::out_of_range & ex) {
+            index = -1;
+        }
+        if (index < 0 || index >= argc) {
+            throw ArgumentParserInvalidValueError(
+                M_("Misconfigured alias \"{}\": index \"{}\" out of range"), std::string(argv[0]), match[0].str());
```

Comment:
The exception message appends `Add "--help" for more information about the arguments` suffix:

~~~~
dnf5 requires 'perl(lib)' --installed
Misconfigured alias "requires": index "${11111111111111111111111111111111111111}" out of range. Add "--help" for more information about the arguments.
~~~~

That suffix is misleading in this case. We should point to dnf5-aliases manual instead (or add no suffix). Should we use a different exception type?